AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java

```
@@ -265,17 +265,51 @@ public boolean onSelection(MaterialDialog materialDialog, View view, int which,
         }
     };
 
-    private MaterialDialog.ListCallback mMySearchDialogListener =
-            new MaterialDialog.ListCallback() {
+    private CardBrowserMySearchesDialog.MySearchesDialogListener mMySearchesDialogListener =
+            new CardBrowserMySearchesDialog.MySearchesDialogListener() {
         @Override
-        public void onSelection(MaterialDialog materialDialog, View view, int which,
-                                   CharSequence text) {
+        public void OnSelection(String searchName) {
             JSONObject savedFiltersObj = getCol().getConf().optJSONObject("savedFilters");
             if (savedFiltersObj != null) {
-                mSearchTerms = savedFiltersObj.optString(text.toString());
+                mSearchTerms = savedFiltersObj.optString(searchName);
                 mSearchView.setQuery(mSearchTerms, false);
-                searchCards();
                 MenuItemCompat.expandActionView(mSearchItem);
+                searchCards();
+            }
+        }
+
+        @Override
+        public void OnSaveNewSearch(String searchName, String searchTerms) {
+            try {
+                JSONObject savedFiltersObj = getCol().getConf().optJSONObject("savedFilters");
+                boolean should_save = false;
+                if (savedFiltersObj == null) {
+                    savedFiltersObj = new JSONObject();
+                    savedFiltersObj.put(searchName, searchTerms);
+                    should_save = true;
+                } else if (!savedFiltersObj.has(searchName)) {
+                    savedFiltersObj.put(searchName, searchTerms);
+                    should_save = true;
+                }
+                if (should_save) {
+                    getCol().getConf().put("savedFilters", savedFiltersObj);
+                    getCol().flush();
+                }
+            } catch (JSONException e) {
+            }
```

Can you throw a new RuntimeException for these? It's harder to track down problems if they fail silently like this.